src/lib.rs

```diff
@@ -1320,8 +1320,9 @@ impl GooseAttack {
         // in the current step.
         debug_assert!(self.test_plan.steps[self.test_plan.current].0 > previous_users);
 
-        // Divide the number of new users to launch by the time configured to launch them.
-        let hatch_rate: f32 = (self.test_plan.steps[self.test_plan.current].0 - previous_users)
+        // Divide the number of new users to launch (minus one, as the first launches immediately without any delay)
```

Comment:
Out of curiosity - why is that the case?

Reply:
It's something we could revisit, but originally it's because when Starting a load test the first GooseUser thread starts immediately, versus waiting whatever hatch-rate milliseconds before starting the first user, as otherwise the load test is idle for some time.

Another idea would be to just add a delay after the last user starts but before we advance to the next step. But this gets a little fuzzy if starting lots of users quickly because users may not start at the exact cadence configured (ie, if the server simply can't keep up with it).